hikari/impl/rate_limits.py

```diff
@@ -129,6 +129,7 @@ def close(self) -> None:
 
         if self.throttle_task is not None:
             self.throttle_task.cancel()
+            asyncio.shield(self.throttle_task)
```

Comment:
might want to set that to None after cancelling it here as well (plus shielding after cancelling?)